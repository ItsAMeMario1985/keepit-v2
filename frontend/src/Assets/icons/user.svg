<svg xmlns="http://www.w3.org/2000/svg" width="13"  viewBox="0 0 19.161 15.329">
  <path id="Icon_awesome-user-edit" data-name="Icon awesome-user-edit" d="M6.706,7.665A3.832,3.832,0,1,0,2.874,3.832,3.832,3.832,0,0,0,6.706,7.665Zm2.683.958h-.5a5.212,5.212,0,0,1-4.365,0h-.5A4.025,4.025,0,0,0,0,12.647v1.245a1.437,1.437,0,0,0,1.437,1.437h8.23a1.438,1.438,0,0,1-.078-.638l.2-1.823.036-.332.237-.237L12.38,9.985A3.983,3.983,0,0,0,9.389,8.623Zm1.356,4.35-.2,1.826a.477.477,0,0,0,.527.527l1.823-.2,4.129-4.129L14.874,8.847l-4.129,4.126Zm8.206-4.922L17.817,6.916a.717.717,0,0,0-1.012,0L15.673,8.048l-.123.123,2.15,2.147,1.251-1.251a.721.721,0,0,0,0-1.015Z" fill="#fff"/>
</svg>
