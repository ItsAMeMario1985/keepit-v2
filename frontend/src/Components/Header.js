import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { ReactComponent as Logo } from '../Assets/logo.svg'

export default function Header({ onClick, buttonText }) {
  return (
    <StyledHeader>
      <Logo></Logo>
    </StyledHeader>
  )
}

const StyledHeader = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 80px;
`
