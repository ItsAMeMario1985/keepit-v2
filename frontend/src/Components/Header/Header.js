import styled from 'styled-components/macro'
import { useHistory } from 'react-router-dom'
import { ReactComponent as Logo } from '../../Assets/logo.svg'

export default function Header() {
  const history = useHistory()
  function handleOnClick() {
    history.push('/')
  }
  return (
    <StyledHeader>
      <Logo data-testid="TestHeaderLogo" width="157" onClick={handleOnClick} />
    </StyledHeader>
  )
}

const StyledHeader = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100px;

  svgÂ  {
    cursor: pointer;
  }
`
