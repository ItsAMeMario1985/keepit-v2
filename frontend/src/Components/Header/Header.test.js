import { render } from '@testing-library/react'
import Header from './Header'
import user from '@testing-library/user-event'

const mockHistoryPush = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

describe('Header', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('onclick works', () => {
    const { getByTestId } = render(<Header />)
    const TestHeaderLogo = getByTestId('TestHeaderLogo')
    user.click(TestHeaderLogo)
    expect(mockHistoryPush).toHaveBeenCalledTimes(1)
  })
})
