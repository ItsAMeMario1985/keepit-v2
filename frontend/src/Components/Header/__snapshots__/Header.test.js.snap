// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="Header__StyledHeader-sc-1kdjcf7-0 hpCfJW"
>
  <svg>
    logo.svg
  </svg>
</header>
`;
